resources/local/podman/vm.go

```diff
@@ -49,14 +50,22 @@ func NewInstance(e resourceslocal.Environment, args VMArgs, opts ...pulumi.Resou
 		return pulumi.StringOutput{}, "", -1, err
 	}
 
+	// Use a config to avoid docker hooks that can call vault or other services (credHelpers)
+	err = os.WriteFile(path.Join(dataPath, "config.json"), []byte(customDockerConfig), 0600)
+	if err != nil {
+		return pulumi.StringOutput{}, "", -1, err
+	}
+
+	podmanCommand := "podman --config " + dataPath
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do look for something to sincerely praise. -->
**👏 praise**
Oh nice!!